common/sema.h

```diff
@@ -205,8 +207,7 @@ class LightweightSemaphore
     void wait()
     {
         if (!tryWait()) {
-            // We will allow spinning only if we know multiple "hardware thread contexts" are available
-            if (std::thread::hardware_concurrency() > 1) {
+            if (s_multicore) {
```

Comment:
You could also put `static bool s_multicore =  std::thread::hardware_concurrency() > 1` right in front of that it, static local variables are [guaranteed](http://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables) to be initialized once.